File: frontend/src/lib/era.ts
// Expansion-era constants, mirroring backend internal/era. Project Quarm is
// pre-Planes-of-Power until the expansion launches (October 2026); the
// pop_enabled preference (Developer tab preview toggle, default off) swaps
// the whole app over. Read the flag via hooks/usePoPEnabled.

export const PRE_POP_MAX_LEVEL = 60
export const POP_MAX_LEVEL = 65

// maxLevel returns the server level cap for the given era state.
export function maxLevel(popEnabled: boolean): number {
  return popEnabled ? POP_MAX_LEVEL : PRE_POP_MAX_LEVEL
}

File: backend/internal/era/era.go
// Package era centralizes Project Quarm's expansion-era gating. The server
// is currently in the pre-Planes-of-Power era (level cap 60, PoP content
// locked); PoP launches later in 2026. Every level cap or era-conditional
// branch in the app should derive from this package so the whole app swaps
// over together when Preferences.PoPEnabled flips — via the Developer-tab
// preview toggle today, and via the default at launch.
//
// Phase B/C of the PoP rollout (content additions, data refresh, default
// flip) builds on this package; see the era-gated call sites for what each
// flag value controls.
package era

const (
	// PrePoPMaxLevel is the level cap while Planes of Power is locked.
	PrePoPMaxLevel = 60
	// PoPMaxLevel is the level cap once Planes of Power is live.
	PoPMaxLevel = 65
)

// MaxLevel returns the server level cap for the given era state. popActive
// is Preferences.PoPEnabled — false until the expansion launches.
func MaxLevel(popActive bool) int {
	if popActive {
		return PoPMaxLevel
	}
	return PrePoPMaxLevel
}
